include/deal.II/physics/transformations.h

```diff
@@ -0,0 +1,1177 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__transformations_h
+#define dealii__transformations_h
+
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/symmetric_tensor.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+namespace Physics
+{
+
+  /**
+  * A collection of operations to assist in the transformation of tensor
+  * quantities from the reference to spatial configuration, and vice versa.
+  * These types of transformation are typically used to reexpress quantities
+  * measured or computed in configuration in terms of a second configuration.
+  *
+  * For convenience we will predefine some commonly referenced tensors and
+  * operations.
+  * Considering the position vector $\mathbf{X}$ in the referential (material)
+  * configuration, points $\mathbf{X}$ are transformed to points $\mathbf{x}$
+  * in the current (spatial) configuration through the nonlinear map
+  * @f[
+  *  \mathbf{x}
+  *   := \boldsymbol{\varphi} \left( \mathbf{X} \right)
+  *    = \mathbf{X} + \mathbf{u}(\mathbf{X})
+  * @f]
+  * where the $\mathbf{u}(\mathbf{X})$ represents the displacement vector.
+  * From this we can compute the deformation gradient tensor as
+  * @f[
+  *  \mathbf{F} := \mathbf{I} + \nabla_{0}\mathbf{u} \, ,
+  * @f]
+  * wherein the differential operator $\nabla_{0}$ is defined as
+  * $\frac{\partial}{\partial \mathbf{X}}$ and $\mathbf{I}$ is the identity
+  * tensor.
+  *
+  * @note For these transformations, unless otherwise stated, we will strictly
+  * assume that all indices of the transformed tensors derive from one coordinate
+  * system; that is to say that they are not multi-point tensors (such as the
+  * Piola stress in elasticity).
+  *
+  * @author Jean-Paul Pelteret, Andrew McBride, 2016
+  */
+  namespace Transformations
+  {
+
+    /**
+     * Transformation of tensors that are defined in terms of a set of
+     * contravariant bases. Rank-1 and rank-2 contravariant tensors
+     * $\left(\bullet\right)^{\#} = \mathbf{T}$ (and its spatial counterpart
```

Comment:
I wouldn't say that the notation is _common_, but its the easiest way to highlight that the tensors that these transformations are performed on are not arbitrary. This notation is from Holzapfel's book, while Wriggers (who doesn't define this collection of transformation in general, but rather only showcases a few specific examples) uses some sub- and super-script indices to denote covariant and contravariant bases. 

If I chose the latter option then I'd have to write everything in index notation and define multiple basis vectors, a convective coordinate system, ... all of which even more confusing (to me, at least).

Would you be happy if I were to make a stronger connection between this and what is presented by Holzapfel?

`\#` -> `\sharp` noted. I'll check to see if this works.